pkg/router/endpoints.go

```diff
@@ -42,3 +43,28 @@ func createNewGame(env *Env) gin.HandlerFunc {
 		})
 	}
 }
+
+func createNewResource(env *Env) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		requestBody := &CreateResourceModel{}
+		err := c.ShouldBindJSON(requestBody)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{
+				"detail": "Invalid request body, expected `GameID, ResourceName`",
+			})
+			return
+		}
+
+		// Try to add a resource. If it's not possible due to a lack of money, log it to the user.
+		err = resources.CreateResource(env.DB, requestBody.GameID, requestBody.ResourceName)
+		if err != nil {
+			env.Log.Errorf("Error when creating resource: %v", err)
+			c.JSON(http.StatusInternalServerError, gin.H{
+				"detail": "Failed to create game.",
+			})
+			return
+		}
+
```

Comment:
Need to set a success context here -> you might just have not got that far yet but I'll leave the comment anyway